pkg/plugins/policies/core/xds/meshroute/clusters.go

```diff
@@ -22,7 +22,7 @@ func GenerateClusters(
 
 	for _, serviceName := range services.Sorted() {
 		service := services[serviceName]
-		protocol := generator.InferProtocol(proxy, service.Clusters())
+		protocol := generator.InferProtocol(meshCtx.ServiceInformations.Protocol, service.Clusters())
```

Comment:
I'm not sure to understand why we infer a common protocol when we build a different cluster for each of the service.

Reply:
Fixed